include/libswiftnav/signal.h

```diff
@@ -61,6 +80,10 @@ static inline bool sid_is_equal(const gnss_signal_t a, const gnss_signal_t b)
   return sid_compare(a, b) == 0;
 }
 
+/** Constructs and returns a gnss_signal_t structure using given parameters.
+ */
+gnss_signal_t construct_sid(enum code code, u16 sat);
```

Comment:
Do we need this function for something in particular? It requires the caller to know in advance the valid sat / code combinations, which I was hoping to abstract away in signal.c